### ISSUE TITLE
Integrate shipmonk/name-collision-detector to catch duplicate FQCNs in stubs

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#954]
## Problem

Stub files are loaded by `Plugin::getCommonStubs()` via `RecursiveDirectoryIterator` over `stubs/common/`. If two stub files declare the same FQCN, Psalm silently lets the last-loaded one win. No build error, no warning. The bug only surfaces downstream when a caller observes wrong types.

Concrete failure mode: a contributor copies `stubs/common/Foo/Bar.phpstub` to `stubs/common/Foo/BarV2.phpstub` to draft a change and forgets to delete the original. CI stays green. The wrong stub ships.

Other layers in the repo are safe by construction:
- `src/`, `tests/Unit/`, `tests/Application/app/` — PSR-4. Composer autoloader catches duplicates.
- `tests/Type/tests/*.phpt` — each runs in its own psalm-tester process; cross-file FQCN reuse is fine.

Only `stubs/` is at risk.

## Proposal

Add `shipmonk/name-collision-detector` as a dev dependency and wire it into CI. Run it scoped to stub directories.

### Scoping rules

The detector must not flag **intentional** overrides:

- `stubs/<version>/` deliberately re-declares classes from `stubs/common/` (per CLAUDE.md: "later versions override earlier ones for same-named stubs").
- `stubs/integrations/<pkg>/` may re-declare third-party classes that don't exist in `stubs/common/`.

Solution: run the detector independently against each directory bucket, not the union.

Buckets:
1. `stubs/common/` — strict, no duplicates allowed.
2. Each `stubs/<version>/` (e.g., `stubs/12.42.0/`) — strict within itself.
3. Each `stubs/integrations/<pkg>/` — strict within itself.

### Configuration

```neon
# name-collision-detector.neon (one per bucket, or single config with multiple scanPaths runs)
scanPaths:
    - stubs/common
fileExtensions:
    - phpstub
```

If a single config is preferred, the CI step loops over buckets and runs the detector per bucket:

```bash
for dir in stubs/common stubs/*/ stubs/integrations/*/; do
    [ -d "$dir" ] || continue
    vendor/bin/detect-collisions --configuration=ci/name-collision.neon "$dir"
done
```

### CI integration

New step in the existing workflow alongside `composer cs`, `composer psalm`, `composer test:unit`:

```yaml
- name: Detect stub FQCN collisions
  run: composer detect:collisions
```

Composer script:

```json
"detect:collisions": "for dir in stubs/common stubs/*/ stubs/integrations/*/; do [ -d \"$dir\" ] && vendor/bin/detect-collisions \"$dir\" || true; done"
```

(Exact form TBD — may need a small wrapper script in `bin/` if shell-in-composer.json gets unwieldy.)

### Allow-list

If a true intentional duplicate is ever needed inside one bucket, the detector supports an ignore list. Add to config with a comment explaining why. Default posture: zero duplicates.

## Out of scope

- Running the detector on `src/` or `tests/` — already covered by Composer autoloader.
- Detecting *stub vs vendor* FQCN overlap (that's the whole point of stubs).

## Acceptance

- Dev dep added to `composer.json`.
- `composer detect:collisions` runs locally and in CI, exits 0 on current `master` / `3.x`.
- A seeded duplicate (e.g., a temporary copy of `stubs/common/Support/Collection.phpstub`) makes the script fail with a clear message.
- README or `docs/contributing/` mentions the check.
